libs/entitlement/qa/common/src/main/java/org/elasticsearch/entitlement/qa/common/NetworkAccessCheckActions.java

```diff
@@ -53,10 +63,145 @@ static void socketBind() throws IOException {
         }
     }
 
-    @SuppressForbidden(reason = "Testing entitlement check on forbidden action")
     static void socketConnect() throws IOException {
         try (Socket socket = new DummyImplementations.DummySocket()) {
             socket.connect(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
         }
     }
+
+    static void serverSocketChannelBind() throws IOException {
+        try (var serverSocketChannel = ServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
+        }
+    }
+
+    static void serverSocketChannelBindWithBacklog() throws IOException {
+        try (var serverSocketChannel = ServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0), 50);
+        }
+    }
+
+    static void serverSocketChannelAccept() throws IOException {
+        try (var serverSocketChannel = ServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
+            serverSocketChannel.configureBlocking(false);
+            serverSocketChannel.accept();
+        }
+    }
+
+    static void asynchronousServerSocketChannelBind() throws IOException {
+        try (var serverSocketChannel = AsynchronousServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
+        }
+    }
+
+    static void asynchronousServerSocketChannelBindWithBacklog() throws IOException {
+        try (var serverSocketChannel = AsynchronousServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0), 50);
+        }
+    }
+
+    static void asynchronousServerSocketChannelAccept() throws IOException {
+        try (var serverSocketChannel = AsynchronousServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
+            var future = serverSocketChannel.accept();
+            future.cancel(true);
+        }
+    }
+
+    static void asynchronousServerSocketChannelAcceptWithHandler() throws IOException {
+        try (var serverSocketChannel = AsynchronousServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
+            serverSocketChannel.accept(null, new CompletionHandler<>() {
+                @Override
+                public void completed(AsynchronousSocketChannel result, Object attachment) {}
+
+                @Override
+                public void failed(Throwable exc, Object attachment) {
+                    assert exc.getClass().getSimpleName().equals("NotEntitledException") == false;
```

Comment:
This assertion is a good idea. Could it be written like this?

```
assert (exc instanceof NotEntitledException) == false;
```

Reply:
I did not want to reference the entitlement lib from the plugin(s)